postgres_composite_types/__init__.py

```diff
@@ -258,10 +259,32 @@ def __init__(cls, name, bases, attrs):
         if name == 'CompositeType':
             return
 
+        cls._capture_descriptors()
+
         # Register the type on the first database connection
         connection_created.connect(receiver=cls.database_connected,
                                    dispatch_uid=cls._meta.db_type)
 
+    def _capture_descriptors(cls):
+        """ Work around for not being able to call contribute_to_class."""
```

Comment:
Can we not do this work in `__new__` when we're constructing the class? Perhaps by building a map of descriptors we need to make sure are set on the class unmodified in `__init__`. That would save needing to manufacture new types.